src/routes/Articles.jsx

```diff
@@ -1,33 +1,45 @@
 import * as React from "react";
-import ArticleCard from "./components/Article-card";
-import { Grid, Container, CircularProgress, Box, Typography } from "@mui/material";
+import { useLocation } from 'react-router-dom';
+import ArticleCard from "./components/ArticleCard";
+import TopicSelector from "./components/TopicSelector";
+import {
+  Container,
+  CircularProgress,
+  Box,
+  Typography,
+} from "@mui/material";
 import newsApi from "../api-calls/axios";
 
 const Articles = () => {
   const [articles, setArticles] = React.useState([]);
   const [loading, setLoading] = React.useState(true);
+  const location = useLocation();
 
   React.useEffect(() => {
+    const searchParams = new URLSearchParams(location.search);
+    const topic = searchParams.get('topic');
+    
+    setLoading(true);
     newsApi
-      .get("/articles")
+      .get("/articles", { params: { topic } })
```

Comment:
I appreciate I may have mentioned this before but you can "wrap" these api calls into functions in your api.js as well as the base url  so they just return the requested data when invoked - they are then reusable and reduces the code in your components 👍